distributed-prover/src/circuits/evm.rs

```diff
@@ -0,0 +1,216 @@
+use ark_std::{end_timer, start_timer};
+use bus_mapping::{circuit_input_builder::FixedCParams, mock::BlockData};
+use eth_types::geth_types::GethData;
+use halo2_proofs::{
+    halo2curves::bn256::{Bn256, Fr, G1Affine},
+    plonk::{
+        create_proof as create_proof_local, distributed_prover::{
+            config:: WorkloadConfig,
+            prover::create_proof as create_proof_distributed,
+        },
+        keygen_pk, keygen_vk, verify_proof,
+        Circuit,
+    },
+    poly::{
+        commitment::ParamsProver,
+        kzg::{
+            commitment::{KZGCommitmentScheme, ParamsKZG},
+            multiopen::{ProverSHPLONK, VerifierSHPLONK},
+            strategy::SingleStrategy,
+        },
+    },
+    transcript::{
+        Blake2bRead, Blake2bWrite, Challenge255, TranscriptReadBuffer, TranscriptWriterBuffer,
+    },
+};
+use mock::TestContext;
+use rand::SeedableRng;
+use rand_xorshift::XorShiftRng;
+use zkevm_circuits::{
+    evm_circuit::{EvmCircuit, witness::block_convert, TestEvmCircuit},
+    util::SubCircuit
+};
+
+use crate::util::artifacts::*;
+use crate::util::constants::RNG_SEED;
+
+pub(crate) const CIRCUIT_NAME: &str = "evm";
+const CIRCUIT_DEGREE: u32 = 18;
+
+// What is the <Fr> notation?
```

Comment:
A generic type can be instantiated with a specific type annotation.

Here, EvmCircuit is the generic type that requires a Field, and Fr is supplied as the specific type.